CMakeLists.txt

```diff
@@ -17,3 +17,31 @@ add_executable(hershensh hershensh.cpp ${HEADER_FILES})
 
 target_include_directories(hershensh 
   PUBLIC "${INCLUDE_DIR}")
+
+#-------------------------
+# Catch2 unittesting framework
+Include(FetchContent)
```

Comment:
1. Use ExternalProject so that building happens in build time rather than configure time.
2. If it doesn't take long to build this (less than 5 seconds) then feel free to merge this if you want and I'll open an issue so we can fix it later.

Reply:
Why do you think that's better?

Reply:
I am not suggesting to configure at compile time. I'm suggesting to build this project in build time and not in configure time.

Reply:
1. I couldn't get it to work with `ExternalProject`, as during configure time, cmake is not aware of the Catch2 project and I get different errors depending on what operations I do after `ExternalProject_Add`. You can try to make it work if you wish, but:
2. `FetchContent` performs the git clone operation at configure time. `ExtrenalProject` performs the git clone at build time. Both do the compilation at build time. It seems more natural to me to do the git clone at configure time so that the project has both its own source files, and the Catch2 source files. Then at build time, both targets are compiled. I still don't understand why you think the git clone should be done at build time. Care to explain?

Reply:
I'm merging so I can work keep working with these changes, but feel free to keep discussing.